deploy/aws-ssm-agent/01-machineconfig.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-install-aws-ssm-agent
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - name: 99-install-aws-ssm-agent.service
+        enabled: true
+        contents: |
+          [Unit]
+          Description=Install and launch the AWS Systems Manager (SSM) Agent.
```

Comment:
(This is a stock OSD cluster I'm testing on, so feel free to discard this if permissions already exist on MCs)
Without the `ssm:UpdateInstanceInformation` capability, you'll get this error in the node logs: 
```
oc adm node-logs <node>

Nov 21 23:17:19.820520 ip-10-0-212-35 amazon-ssm-agent[5720]: 2023-11-21 23:17:17 WARN EC2RoleProvider Failed to connect to Systems Manager with instance profile role credentials. Err: retrieved credentials failed to report to ssm. RequestId: ef59672d-9050-4b02-83fb-5d028be3f603 Error: AccessDeniedException: User: arn:aws:sts::794240594095:assumed-role/hkemp-muo-ocm-pwpm7-worker-role/i-04c463472fe394135 is not authorized to perform: ssm:UpdateInstanceInformation on resource: arn:aws:ec2:us-west-2:794240594095:instance/i-04c463472fe394135 because no identity-based policy allows the ssm:UpdateInstanceInformation action

Nov 21 23:17:19.820520 ip-10-0-212-35 amazon-ssm-agent[5720]:         status code: 400, request id: ef59672d-9050-4b02-83fb-5d028be3f603
Nov 21 23:17:19.920810 ip-10-0-212-35 amazon-ssm-agent[5720]: 2023-11-21 23:17:17 ERROR EC2RoleProvider Failed to connect to Systems Manager with SSM role credentials. error calling RequestManagedInstanceRoleToken: AccessDeniedException: Systems Manager's instance management role is not configured for account: 794240594095

Nov 21 23:17:19.920810 ip-10-0-212-35 amazon-ssm-agent[5720]:         status code: 400, request id: 1d9f50e4-f035-43c8-b6fc-26c08bd7f86e
Nov 21 23:17:20.021220 ip-10-0-212-35 amazon-ssm-agent[5720]: 2023-11-21 23:17:17 ERROR [CredentialRefresher] Retrieve credentials produced error: no valid credentials could be retrieved for ec2 identity. Default Host Management Err: error calling RequestManagedInstanceRoleToken: AccessDeniedException: Systems Manager's instance management role is not configured for account: 794240594095
```

Reply:
Yeah, the IAM permissions will be taken care of separately 

https://issues.redhat.com/browse/OSD-19652 for the instance profile and
https://issues.redhat.com/browse/OSD-19653 for SREs to be able to interact with SSM